test/test_torch.py

```diff
@@ -1927,56 +1942,50 @@ def check_sum_all(tensor):
     def _assert_matches_numpy(self, t, n):
         self.assertEqual(n.shape, t.shape)
         if t.dtype == torch.float:
-            self.assertTrue(np.allclose(n, t.numpy(), rtol=1e-03, atol=1e-05))
+            self.assertTrue(np.allclose(n, t.numpy(), rtol=1e-03, atol=1e-05,
+                            equal_nan=True))
         else:
-            self.assertTrue(np.allclose(n, t.numpy()))
+            self.assertTrue(np.allclose(n, t.numpy(), equal_nan=True))
+
+    def _test_dim_op(self, tensors_dict, dim, pytorch_op, numpy_op):
+        for category, tensors in tensors_dict.items():
+            if category == "slice":
+                dim = 0
+            for tensor in tensors:
+                # we have no control over NumPy warnings...
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    expected = numpy_op(tensor.numpy(), dim)
+                actual = pytorch_op(tensor, dim)
+                self._assert_matches_numpy(actual, expected)
 
     @unittest.skipIf(not TEST_NUMPY, 'Numpy not found')
     def test_sum_dim(self):
-        def check_sum_dim(tensors_dict, dim):
-            for category, tensors in tensors_dict.items():
-                if category == "slice":
-                    dim = 0
-                for tensor in tensors:
-                    expected = tensor.numpy().sum(dim)
-                    actual = tensor.sum(dim)
-                    self._assert_matches_numpy(actual, expected)
-
-        float_types = [torch.double, torch.float]
-        int_types = [torch.int64, torch.int32, torch.int16]
-
-        check_sum_dim(self._make_tensors((5, 400000)), 1)
-        check_sum_dim(self._make_tensors((3, 5, 7)), 0)
-        check_sum_dim(self._make_tensors((3, 5, 7)), 1)
-        check_sum_dim(self._make_tensors((3, 5, 7)), 2)
-        check_sum_dim(self._make_tensors((100000, )), -1)
-        check_sum_dim(self._make_tensors((50, 50, 50)), 0)
-        check_sum_dim(self._make_tensors((50, 50, 50)), 1)
-        check_sum_dim(self._make_tensors((50, 50, 50)), 2)
-        check_sum_dim(self._make_tensors((50, 50, 50)), (1, 2))
-        check_sum_dim(self._make_tensors((50, 50, 50)), (1, -1))
+        for sizes, dim in DIM_TEST_SCENARIOS:
```

Comment:
The reason for the unrolled pattern in the previous code is that it makes it easier to figure out which test case broke in a test failure. With the for-loop, the stack trace doesn't give any information about the failing data.